### ISSUE TITLE
More views

### ISSUE THREAD

**Issue by gallais** [jfdm/tears-in-the-rain#1]
Dumping this here so I don't forget it. Probably worth inserting them before the big one
on linear decidable equality.

# How do you program by induction over `LT` defined in terms of `LTE`?

The answer is in `base`: use a view on `LTE` that exposes an `LT` subterm in the `Succ` case!

https://github.com/idris-lang/Idris2/blob/21f2913527a3911a14f54641e5bc844cb71618cd/libs/base/Data/Nat.idr#L87-L102

# How do you get a catchall to fire on an infinite primitive type

You cannot list all possible alternatives. What do you do? Here is Agda code I just wrote
to answer a question like this on zulip:

```agda
data View : Char -> Set where
  a : View 'a'
  b : View 'b'
  c : View 'c'

view : (c : Char) -> Maybe (View c)
view 'a' = just a
view 'b' = just b
view 'c' = just c
view _ = nothing

dec : Char -> Fin 4
dec c with view c
... | just a = # 0
... | just b = # 1
... | just c = # 2
... | nothing = # 3
```

**Comment by jfdm** [comment]
Indeed! I am currently adding Views and Covers. so that can help with the lead up to linear decidable equality.

**Comment by jfdm** [comment]
I have added a solution documenting ['views & covering functions'](https://jfdm.github.io/tears-in-the-rain/solution/views.html) there is [one for predicates too](https://jfdm.github.io/tears-in-the-rain/solution/predicates.html).  

Both of these solutions are great examples in using views, and they should both be added to the section on views.

I agree that both problems are interesting to add as separate problems.
+ The first one  can be described more generically as: How do you program over predicates defined as instances (using type-synonyms) of existing ones? The solution: Views!
+ The second one looks reasonable as well.

**Comment by gallais** [comment]
* `PairOdd` seems like a strange name (it does not imply that the list has an odd length)
* `IsEmpty` is a confusing name as it's characterising precisely the non-empty lists

**Comment by jfdm** [comment]
I have fixed those minor issues you pointed out with `PairOdd` and `IsEmpty`.
